src/ast/passes/semantic_analyser.cpp

```diff
@@ -3380,6 +3431,16 @@ void SemanticAnalyser::visit(Expression &expr)
     } else {
       pass_tracker_.inc_num_unresolved();
     }
+  } else if (auto *szof = expr.as<Sizeof>()) {
+    const auto v = check(*szof);
+    if (v) {
+      expr.value = ctx_.make_node<Integer>(*v, Location(szof->loc));
```

Comment:
Can we force this to be a uint64 so the API doesn't change?

Reply:
It is weird. We have:

```
if 8 == sizeof(int64) { ... }
```

This would generate a signed/unsigned integer comparison in some cases. I am not clear on why `5` is a signed integer by default, but this is the legacy contract. Do you think we can change signed literals to be `uint64` by default if they are unsigned?

I'm sure we can come up with something here. I think you're suggesting that the default type just for `offsetof` and `sizeof` remains unsigned? I could maybe add a type-hint to `Integer` or something similar.

Reply:
This was actually really simple. The constructor just has a `force_unsigned` parameter, which overrides the existing behavior. The `sizeof` and `offsetof` types will be typed like this by default. But if you do `sizeof(int8) + sizeof(int8)` it will be a signed integer, matching the current semantics.

Reply:
> But if you do sizeof(int8) + sizeof(int8) it will be a signed integer, matching the current semantics.

Huh? Why would the addition of two unsigned ints equal a signed int. Currently, `$x` in this program is an unsigned int: `$x = sizeof(int8) + sizeof(int8);`